src/VisualStudio/IntegrationTest/TestUtilities/InProcess/DialogHelpers.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Automation;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess
+{
+    public static class DialogHelpers
+    {
+        public static AutomationElement FindDialog(int visualStudioHWnd, string dialogAutomationName, bool isOpen)
+        {
+            return Retry(
+                () => FindDialogWorker(visualStudioHWnd, dialogAutomationName),
+                stoppingCondition: automationElement => isOpen ? automationElement != null : automationElement == null,
+                delay: TimeSpan.FromMilliseconds(250));
+        }
+
+        private static AutomationElement FindDialogWorker(int visualStudioHWnd, string dialogAutomationName)
+        {
+            var vsAutomationElement = AutomationElement.FromHandle(new IntPtr(visualStudioHWnd));
+
+            Condition elementCondition = new AndCondition(
+                new PropertyCondition(AutomationElement.AutomationIdProperty, dialogAutomationName),
+                new PropertyCondition(AutomationElement.ControlTypeProperty, ControlType.Window));
+
+            return vsAutomationElement.FindFirst(TreeScope.Descendants, elementCondition);
+        }
+
+        private static T Retry<T>(Func<T> action, Func<T, bool> stoppingCondition, TimeSpan delay)
```

Comment:
Not sure why this is living in "DialogHelpers", since it's used a lot of other places too, right?